internal/livereload/connection.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2015 The Hugo Authors. All rights reserved.
```

Comment:
Can we document the diff-patch we do here? This might make it easier to compare with upstream in the future.

Reply:
I left a few comments describing where I took the code from and what I changed :)